### DISCUSSION TITLE
Firmware-Versionen

### DISCUSSION THREAD

**Discussion by emufan** [SukramJ/aiohomematic#830] [Q&A] [answered]
Weiß jemand. nach welcher Regel werden die Firmware-Versionen der HMIP-Geräte in HA angelegt/aktualisiert werden? Ich habe mehrere Geräte, die in HA den Stand haben, zu dem sie in der CCU hinzugefügt worden sind, aber nicht den, den sie derzeit haben:

![image](https://user-images.githubusercontent.com/22775515/211355269-2a263c72-c95f-4501-8f37-e41eb870929b.png)
vs
![image](https://user-images.githubusercontent.com/22775515/211355299-ef30dcb5-1c5f-4509-87ec-a5d38ac1fae0.png)

oder

![image](https://user-images.githubusercontent.com/22775515/211355359-0feaed8d-df44-47c8-8171-ee187d875e14.png)
vs
![image](https://user-images.githubusercontent.com/22775515/211355398-449f8ab1-6303-43d2-836b-d6a3c1c6c80c.png)

Zweiter Fall ist heute aktualisiert, da war mir das initial aufgefallen. Der erste wurde aber tatsächlich schon vor Wochen aktualisiert und die CCU und HA auch schon mehrfach durchgestartet seitdem.

**Comment by SukramJ** [comment, marked as answer]
Da die Firmware Version nur in der `device_description` liegt wird der Wert nur bei der Deviceerstellung in HA aktualisiert und gecached.

Um diese Werte zu aktualisieren muss der Cache gelöscht werden: `Homematic(IP) Local: Clear the cache`

**Reply by emufan** [reply]
Ah. o.k. Danke. Ist mir nicht so wichtig, aber wollte ich dennoch verstehen.

**Comment by ProfDrYoMan** [comment]
Kann man das vllt so erweitern, dass eine update.device entity angelegt wird?

**Reply by SukramJ** [reply]
Ist das bei Homematic sinnvoll?
Man lädt über die CCU die Dateien hoch, dann dauert es ewig bis die Firmware auf das Gerät übertragen wurde, und zuletzt gibt es, soweit ich weiss keine API Methode um das Update auf dem Gerät durchzuführen.

**Reply by ProfDrYoMan** [reply]
Das ist korrekt, man kann das nicht über HA machen, aber die Info in HA wäre schon fein.
Ich hab die CCU GUI einmal alle halbe Jahr offen. :) Naja, wenn dann ein update kommt, könnte ich es auch ertragen. War nur ne Idee.

**Reply by SukramJ** [reply]
Ich habe leider keine Ahnung, wie ich an diese Informationen kommen könnte.

**Reply by ProfDrYoMan** [reply]
Dann ists ja klar, dass es einfach nicht gemacht wird. :) Auch gut.
